samza-core/src/main/java/org/apache/samza/clustermanager/ResourceRequestState.java

```diff
@@ -46,16 +48,24 @@
    * Maintain a map of hostname to a list of resources allocated on this host
    */
   private final Map<String, List<SamzaResource>> allocatedResources = new HashMap<>();
+
   /**
    * Represents the queue of resource requests made by the {@link ContainerProcessManager}
    */
-  private final PriorityQueue<SamzaResourceRequest> requestsQueue = new PriorityQueue<SamzaResourceRequest>();
+  private final PriorityQueue<SamzaResourceRequest> requestsQueue = new PriorityQueue<>();
+
+  /**
+   * Represents the queue of delayed resource requests made by the {@link ContainerProcessManager}
```

Comment:
maybe add to documentation: "The difference between requestsQueue and delayedRequestsQueue is that, any request present the requestsQueue has been sent out to the YARN-RM, 
while requests in the delayedRequestsQueue will be sent to the YARN-RM only when their delay reaches 0."